pkg/devfile/parser/parse.go

```diff
@@ -90,83 +102,66 @@ func ParseFromData(data []byte) (d DevfileObj, err error) {
 		return d, err
 	}
 
-	return parseDevfile(d)
+	return parseDevfile(d, true)
 }
 
-func parseParent(d DevfileObj) error {
-	parent := d.Data.GetParent()
-
-	parentData, err := ParseFromURL(parent.Uri)
-	if err != nil {
-		return err
-	}
-	klog.V(4).Infof("overriding data of devfile with URI: %v", parent.Uri)
+func parseParentAndPlugin(d DevfileObj) (err error) {
+	flattenedParent := &v1.DevWorkspaceTemplateSpecContent{}
+	if d.Data.GetParent() != nil {
+		if !reflect.DeepEqual(d.Data.GetParent(), &v1.Parent{}) && d.Data.GetParent().Uri != "" {
+			parent := d.Data.GetParent()
 
-	// override the parent's components, commands, projects and events
-	err = parentData.OverrideComponents(d.Data.GetParent().Components)
-	if err != nil {
-		return err
-	}
+			parentData, err := ParseFromURL(parent.Uri)
```

Comment:
One thing to think about is, you are calling func `ParseFromURL` with both parent and plugin. We need to avoid infinite loops if there the parent or plugin references back to the main devfile. Do we allow such situations acc to the schema?

because it is `parseDevfile(d, true)` inside func `ParseFromURL`

Reply:
it would error out since reference back would result in duplication in flattened devfile?